cupyx/scipy/sparse/compressed.py

```diff
@@ -558,13 +644,253 @@ def _get_major_slice(self, major):
+            if s is None:
+                raise IndexError('newaxis is not supported')
+            elif s is Ellipsis:
+                ellipsis = i
+                n_ellipsis += 1
+        if n_ellipsis > 0:
+            ellipsis_size = self.ndim - (len(slices) - 1)
+            slices[ellipsis:ellipsis + 1] = [slice(None)] * ellipsis_size
+
+        if len(slices) == 2:
+            row, col = slices
+        elif len(slices) == 1:
+            row, col = slices[0], slice(None)
+        else:
+            raise IndexError('invalid number of indices')
+
+        major, minor = self._swap(row, col)
+        major_size, minor_size = self._swap(*self._shape)
+
+        if type(major) is slice or type(minor) is slice:
+            raise ValueError("Unsupported assignment")
+        else:
+            if type(major) is int:
+                major = [major]
+            if type(minor) is int:
+                minor = [minor]
+            # validate boundaries
+            if max(major) >= major_size or min(major) < - major_size \
+                    or max(minor) >= minor_size or min(minor) < - minor_size:
+                raise IndexError('index out of bounds')
+            # flatten rows, cols, values lists
+            major = cupy.array(major, dtype=self.indices.dtype,
+                               copy=False, ndmin=1).ravel()
+            minor = cupy.array(minor, dtype=self.indices.dtype,
+                               copy=False, ndmin=1).ravel()
+            values = cupy.array(values, dtype=self.dtype,
+                                copy=False, ndmin=1).ravel()
+            self._set_items(major, minor, values, major_size, minor_size)
+
+    def _set_items(self, major, minor, values, major_size, minor_size):
+        """
+         Assigns multiple values to the required positions.
+         If the position is found, the value is updated (whether 0 or not)
+         else the value is inserted
+         (only if a non zero value is found in the value list)
+         The zeros are eliminated at the end to keep
+         the sparse matrix structure
+
+        Args:
+            major: flattened cupy array of rows indices
+            minor: flattened cupy array of columns indices
+            values: flattened cupy array of values
+            major_size: number of rows
+            minor_size: number of columns
+
+        Returns: void
+
+        """
+        # offsets contain the index of the entry to be updated
+        # if found, else -1
+        offsets = cupy.empty(values.size, dtype=self.indices.dtype)
+        repeat = self.sample_offsets(major_size, minor_size,
+                                     values.size, major, minor, offsets)
+        # In case of duplicates entries, eliminate entries by summation
+        # and repeat the operation
+        if repeat == 1:
+            self.sum_duplicates()
+            self.sample_offsets(major_size, minor_size,
+                                values.size, major, minor, offsets)
```

Comment:
Why is this necessary?

Reply:
So do you suggest to remove it and set duplicate entries each time when found ?

Reply:
I eliminated that part